codalab/client/local_bundle_client.py

```diff
@@ -167,12 +167,16 @@ def get_bundle_target(self, target):
         return (self.model.get_bundle(bundle_uuid), subpath)
 
     def get_worksheet_uuid(self, base_worksheet_uuid, worksheet_spec):
-        if worksheet_spec == '' or worksheet_spec == worksheet_util.HOME_WORKSHEET:
```

Comment:
The previous logic won't work. It will lead to infinite recursion when it comes to dashboard. So my current the logic is the following:

Try to get worksheet uuid and return it.
catch error (it means the worksheet does not exist):
if the requested worksheet is home_worksheet or dashboard, create it
otherwise, just raise that error. 

We can also talk in person :)